eth/api.go

```diff
@@ -1376,6 +1376,152 @@ func (api *PublicDebugAPI) SeedHash(number uint64) (string, error) {
 	return fmt.Sprintf("0x%x", hash), nil
 }
 
+
+type StructLogRes struct {
+	Pc      uint64            `json:"Pc"`
+	Op      string            `json:"Op"`
+	Gas     *big.Int          `json:Gas`
+	GasCost *big.Int          `json:GasCost`
+	Error   error             `json:Error`
+	Stack   []string          `json:"Stack"`
+	Memory  map[string]string `json:Memory`
+	Storage map[string]string `json:Storage`
+}
+
+type TransactionExecutionRes struct {
+	Gas         *big.Int       `json:Gas`
+	ReturnValue string         `json:ReturnValue`
+	StructLogs  []StructLogRes `json:StructLogs`
+}
+
+func (s *PublicDebugAPI) doReplayTransaction(txHash common.Hash) ([]vm.StructLog, []byte, *big.Int, error) {
+	// Retrieve the tx from the chain
+	txData, err := s.eth.ChainDb().Get(txHash.Bytes())
+	tx := new(types.Transaction)
+
+	if err == nil && len(txData) > 0 {
+		if err := rlp.DecodeBytes(txData, tx); err != nil {
+			return nil, nil, nil, err
+		}
+	} else {
+		return nil, nil, nil, fmt.Errorf("Transaction not found or not yet mined")
+	}
+
+	// Retrieve the block details
+	_, blockIndex, _, err := getTransactionBlockData(s.eth.ChainDb(), txHash)
```

Comment:
When using the above function you can skip all this as well. That function contains meta data, including the hash of the block it was included in as well as the index.